src/containers/SecurityDetails.js

```diff
@@ -0,0 +1,293 @@
+      return (
+        <div>
+          <input onFocus={(event) => {
+            event.stopPropagation()
+          }} onClick={(event) => {
+            event.stopPropagation()
+          }}
+                 id={attributeName} placeholder='New Attribute'/>
+          <input onFocus={(event) => {
+            event.stopPropagation()
+          }} onClick={(event) => {
+            event.stopPropagation()
+          }}
+                 id={attributeValue} placeholder='Value'/>
+           <FloatingActionButton mini={true}  style={addButtonStyle}
+                         label='Add'
+                         onClick={(event) => {
+                           event.stopPropagation();
+                           if (parent) {
+                             newAttribute[parent + '.' + document.getElementById(attributeName).value] =
+                               document.getElementById(attributeValue).value;
+                           } else {
+                             newAttribute[document.getElementById(attributeName).value] =
+                               document.getElementById(attributeValue).value;
+                           }
+                           this.dataChanged(this.props.entity.id, this.props.entityType.replace('/', ''), newAttribute);
+                           document.getElementById(attributeName).value = '';
+                           document.getElementById(attributeValue).value = '';
+                         }}>
+                       <ContentAdd />
+          </FloatingActionButton>
+
+        </div>
+      )
+    }
+    return null;
+  }
+
+  isAdmin(id) {
+    for (var i in this.props.entityList) { //Get the right entity from the entity list
+      var e = this.props.entityList[i];
+      if(e.id === id && e.type.replace('/', '') === 'user' ? e : undefined)
+        return e.role === 'admin';
+    }
+    return false;
+  }
+
+  renderAttributeActions(id, type, attribute) {
+    var actionButtons = [];
+    if (this.canWrite(attribute)) {
+      actionButtons.push(this.getDeleteButton({id: id, type: type, attribute: attribute}));
+    }
+    return (<div>{actionButtons}</div>);
+  }
+
+  dataChanged(id, type, data) {
+    var params = {};
+    params.entityId = id;
+    params.entityType = type;
+    for (var attribute in data) {
+      if (data.hasOwnProperty(attribute)) {
+        params.attributeType = attribute;
+        try {
+          params.attributeValue = JSON.parse(data[attribute]);
+        } catch (err) {
+          params.attributeValue = data[attribute];
+        }
+      }
+    }
+    this.props.setEntityAttributes(params);
+  }
+
+  customValidateText(text) {
+    return (text.length > 0);
+
+  }
+
+  isPrimitive(object) {
+    return typeof object !== 'object';
+  }
+
+  stringifyDetail(value, attribute, event) {
+    event.stopPropagation();
+    if (!this.isPrimitive(value)) {
+      var checkExist = setInterval(function () {
+        if (document.getElementsByClassName('editing').length) {
```

Comment:
With react there should be very few cases that you need to use actual dom methods like `.document`, you should only be concerned with the virtual dom. 

I'd refactor this by storing a `editing` field in the state object, and keeping track of which field is being edited that way.